mne/epochs.py

```diff
@@ -1928,37 +1948,72 @@ def as_type(self, ch_type='grad', mode='fast'):
         return _as_meg_type_inst(self, ch_type=ch_type, mode=mode)
 
 
-def _check_baseline(baseline, tmin, tmax, sfreq):
-    """Check for a valid baseline."""
-    if baseline is not None:
-        if not isinstance(baseline, tuple) or len(baseline) != 2:
-            raise ValueError('`baseline=%s` is an invalid argument, must be '
-                             'a tuple of length 2 or None' % str(baseline))
-        # check default value of baseline and `tmin=0`
-        if baseline == (None, 0) and tmin == 0:
-            raise ValueError('Baseline interval is only one sample. Use '
-                             '`baseline=(0, 0)` if this is desired.')
-
-        baseline_tmin, baseline_tmax = baseline
-        tstep = 1. / float(sfreq)
-        if baseline_tmin is None:
-            baseline_tmin = tmin
-        baseline_tmin = float(baseline_tmin)
-        if baseline_tmax is None:
-            baseline_tmax = tmax
-        baseline_tmax = float(baseline_tmax)
-        if baseline_tmin < tmin - tstep:
-            raise ValueError(
-                "Baseline interval (tmin = %s) is outside of epoch "
-                "data (tmin = %s)" % (baseline_tmin, tmin))
-        if baseline_tmax > tmax + tstep:
-            raise ValueError(
-                "Baseline interval (tmax = %s) is outside of epoch "
-                "data (tmax = %s)" % (baseline_tmax, tmax))
-        if baseline_tmin > baseline_tmax:
-            raise ValueError(
-                "Baseline min (%s) must be less than baseline max (%s)"
-                % (baseline_tmin, baseline_tmax))
+def _check_baseline(baseline, tmin, tmax, sfreq,
+                    on_baseline_outside_data='raise'):
```

Comment:
Rather than do all the `tmin/tmax/sfreq` gymnastics, can we just pass `self.times`?

Reply:
We now expect `self.times` instead of `tmin, tmax` (ad21ff9); getting rid of `sfreq` is not as trivial: I tried simply replacing it with `times[-1] - times[0]`, but this breaks a few tests. So I'm not touching that one for now… 

Reply:
Haha or maybe I didn't. Just tried again and all tests do pass. Maybe I actually tried with `times[-1] - times[0]` initially, instead of `times[1] - times[0]`. At any rate, dropped the sfreq param now in a2ea0b5

Reply:
I've reverted this change again, as it won't work with data that only has one time point (as used in `test_pick_seeg_ecog()`)